server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/InheritanceAction.java

```diff
@@ -70,26 +66,21 @@ public void define(NewController context) {
       .setHandler(this)
       .setResponseExample(getClass().getResource("example-inheritance.json"));
 
-    QProfileIdentificationParamUtils.defineProfileParams(inheritance, languages);
+    QProfileRef.defineParams(inheritance, languages);
   }
 
   @Override
   public void handle(Request request, Response response) throws Exception {
-    DbSession session = dbClient.openSession(false);
+    DbSession dbSession = dbClient.openSession(false);
     try {
-      String profileKey = QProfileIdentificationParamUtils.getProfileKeyFromParameters(request, profileFactory, session);
-      QualityProfileDto profile = dbClient.qualityProfileDao().selectByKey(session, profileKey);
-      if (profile == null) {
-        throw new NotFoundException(String.format("Could not find a quality profile with key %s", profileKey));
-      }
-
-      List<QProfile> ancestors = profileLookup.ancestors(profile, session);
-      List<QualityProfileDto> children = dbClient.qualityProfileDao().selectChildren(session, profileKey);
+      QualityProfileDto profile = profileFactory.find(QProfileRef.from(request), dbSession);
```

Comment:
here's an example we'd want to have only one open db connection. Don't you think? If this requirement has changed (to try to have one connection by HTTP request) I'd be interested to know